File: pset1/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{
    // Ask user for input string
    char *input = get_string("Text: ");

    // Loop over input
    float letters = 0;
    float words = 0;
    float sentences = 0;
    for (int i = 0, n = strlen(input); i < n; i++)
    {
        // Check for letters
        if ((input[i] >= 'a' && input[i] <= 'z') || (input[i] >= 'A' && input[i] <= 'Z'))
        {
            letters++;
        }
        // Check for words
        if (input[i] == ' ')
        {
            words++;
        }
        // Check for sentences
        if (input[i] == '!' || input[i] == '?' || input[i] == '.')
        {
            sentences++;
        }
    }
    // Fix words
    words++;
    //  Average number of letters per 100 words
    float L = (100 / words) * letters;
    // Average number of sentences per 100 words
    float S = (100 / words) * sentences;

    /* printf("letters: %f\n", letters);
    printf("words: %f\n", words);
    printf("sentences: %f\n", sentences);
    printf("L: %f\n", L);
    printf("S: %f\n", S); */

    // Return Coleman-Liau index formula
    float formula = 0.0588 * L - 0.296 * S - 15.8;
    // Check max grade
    if (formula < 16 && formula >= 1)
    {
        printf("Grade %.0f\n", formula);
    }
    else if (formula < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }
}

File: pset6/readability/readability.py
from cs50 import get_string

# Ask user for input string
userInput = get_string("Text: ")

# Loop over input
letters = 0
words = 0
sentences = 0
i = 0
n = len(userInput)

while i < n:
    # Check for letters
    if ((userInput[i] >= 'a' and userInput[i] <= 'z') or (userInput[i] >= 'A' and userInput[i] <= 'Z')):
        letters += 1
    # Check for words
    if (userInput[i] == ' '):
        words += 1
    # Check for sentences
    if (userInput[i] == '!' or userInput[i] == '?' or userInput[i] == '.'):
        sentences += 1
    i += 1

# Fix words
words += 1

#  Average number of letters per 100 words
L = (100 / words) * letters
# Average number of sentences per 100 words
S = (100 / words) * sentences

# Return Coleman-Liau index formula
formula = 0.0588 * L - 0.296 * S - 15.8

# Check max grade
if formula < 16 and formula >= 1:
    print(f"Grade {round(formula)}")
elif (formula < 1):
    print("Before Grade 1\n")
else:
    print("Grade 16+\n")